Tests/IntegrationTests/Income/GetIncomeByBillingPartyTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Net;
+using Domain.Entity;
+using IntegrationTests.Abstractions;
+using Newtonsoft.Json;
+using QueryContracts.income;
+
+namespace IntegrationTests.Income;
+
+public class GetIncomeByBillingPartyTests : BaseIntegrationTest {
+    public GetIncomeByBillingPartyTests(IntegrationTestsWebAppFactory application) : base(application) {
+    }
+
+    [Fact]
+    public async Task GetIncomeByBillingParty_NoToken_Fails() {
+        Guid id = Guid.NewGuid();
+        var request = new HttpRequestMessage(HttpMethod.Get, $"api/billing-party/{id.ToString()}/incomes");
+        // Act
+        HttpResponseMessage response = await Client.SendAsync(request);
+
+        // Assert that it is unauthorized
+        Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+    }
+
+    [Fact]
+    public async Task GetIncomeByBillingParty_NoBillingParty_ReturnsAnEmptyList() {
```

Comment:
If there is no billing party should not it throw error ?

Reply:
I guess it depends on our decisions.
I mean some apis tend to return empty lists which also make sense.

In our case, I didnt want to make one extra database call to verify that bp exists , since we dont need to retrieve bp to get incomes. But if you would prefer it to throw an error, i can change it to do so. 👍